WalletWasabi/Services/WalletService.cs

```diff
@@ -321,85 +322,89 @@ public async Task InitializeAsync(CancellationToken cancel)
+			foreach (BlockState blockState in KeyManager.GetTransactionIndex())
+			{
+				var relevantTransactions = confirmedTransactions.Where(x => x.BlockHash == blockState.BlockHash).ToArray();
+				var block = await GetOrDownloadBlockAsync(blockState.BlockHash, cancel);
+				await ProcessBlockAsync(blockState.BlockHeight, block, blockState.TransactionIndices, relevantTransactions);
+			}
 
-								var cnt = 0;
-								foreach (var tx in unconfirmedTransactions)
-								{
-									if (mempoolHashes.Contains(tx.GetHash().ToString().Substring(0, compactness)))
-									{
-										tx.SetHeight(Height.Mempool);
-										await ProcessTransactionAsync(tx);
-										Mempool.TransactionHashes.TryAdd(tx.GetHash());
+			// Go through the filters and queue to download the matches.
+			await BitcoinStore.IndexStore.ForeachFiltersAsync(async (filterModel) =>
+			{
+				if (filterModel.Filter != null) // Filter can be null if there is no bech32 tx.
+				{
+					await ProcessFilterModelAsync(filterModel, cancel);
+				}
+			}, new Height(bestKeyManagerHeight.Value + 1));
+		}
 
-										Logger.LogInfo<WalletService>($"Transaction was successfully tested against the backend's mempool hashes: {tx.GetHash()}.");
-										cnt++;
-									}
-								}
+		private async Task LoadDummyMempoolAsync(SmartTransaction[] unconfirmedTransactions)
+		{
+			try
+			{
+				using (await TransactionProcessingLock.LockAsync())
+				using (var client = new WasabiClient(Synchronizer.WasabiClient.TorClient.DestinationUriAction, Synchronizer.WasabiClient.TorClient.TorSocks5EndPoint))
+				{
+					var compactness = 10;
 
-								if (cnt != unconfirmedTransactions.Length)
-								{
-									SerializeTransactionCache();
-								}
-							}
-						}
-						catch
+					var mempoolHashes = await client.GetMempoolHashesAsync(compactness);
+
+					var cnt = 0;
+					foreach (var tx in unconfirmedTransactions)
+					{
+						if (mempoolHashes.Contains(tx.GetHash().ToString().Substring(0, compactness)))
 						{
-							// When there's a connection failure do not clean the transactions, add it to processing.
-							foreach (var tx in unconfirmedTransactions)
-							{
-								tx.SetHeight(Height.Mempool);
-								await ProcessTransactionAsync(tx);
-								Mempool.TransactionHashes.TryAdd(tx.GetHash());
-							}
+							tx.SetHeight(Height.Mempool);
+							await ProcessTransactionAsync(tx);
+							Mempool.TransactionHashes.TryAdd(tx.GetHash());
 
-							throw;
+							Logger.LogInfo<WalletService>($"Transaction was successfully tested against the backend's mempool hashes: {tx.GetHash()}.");
+							cnt++;
 						}
 					}
+
+					if (cnt != unconfirmedTransactions.Length)
+					{
+						SerializeTransactionCache();
+					}
 				}
-				catch (Exception ex)
-				{
-					Logger.LogWarning<WalletService>(ex);
-				}
-				finally
+
+			}
+			catch (Exception ex)
+			{
+				// When there's a connection failure do not clean the transactions, add it to processing.
```

Comment:
Add it or add `them`?

Reply:
This is copy paste but I will fix that grammatical error